test/integration/targets/iam_user/tasks/main.yml

```diff
@@ -0,0 +1,119 @@
+---
+- name: set up aws connection info
+  module_defaults:
+    group/aws:
+      aws_access_key: "{{ aws_access_key }}"
+      aws_secret_key: "{{ aws_secret_key }}"
+      security_token: "{{ security_token | default(omit) }}"
+      region: "{{ aws_region }}"
+  block:
+
+  - name: ensure ansible user exists
+    iam_user:
+      name: '{{ test_user }}'
+      state: present
+
+  - name: get info on IAM user(s)
+    iam_user_info:
+    register: iam_user_info
+
+  - name: assert correct number of users on path are returned
+    assert:
+      that:
+        - iam_user_info.iam_users | length != 0
+
+  - name: get info on IAM user(s) with name
+    iam_user_info:
+      name: '{{ test_user }}'
+    register: iam_user_info
+  - assert:
+      that:
+        - iam_user_info.iam_users | length == 1
+        - 'iam_user_info.iam_users[0].arn.startswith("arn:aws:iam")'
+        - 'iam_user_info.iam_users[0].arn.endswith("user/" + test_user )'
+        - iam_user_info.iam_users[0].createDate is not none
+        - iam_user_info.iam_users[0].path == '{{ test_path }}'
+        - iam_user_info.iam_users[0].user_id is not none
+        - iam_user_info.iam_users[0].user_name == '{{ test_user }}'
+
+  - name: get info on IAM user(s) on path
+    iam_user_info:
+      path: '{{ test_path }}'
+      name: '{{ test_user }}'
+    register: iam_user_info
+  - assert:
+      that:
+        - iam_user_info.iam_users | length == 1
+        - 'iam_user_info.iam_users[0].arn.startswith("arn:aws:iam")'
+        - 'iam_user_info.iam_users[0].arn.endswith("user/" + test_user )'
+        - iam_user_info.iam_users[0].createDate is not none
+        - iam_user_info.iam_users[0].path == '{{ test_path }}'
+        - iam_user_info.iam_users[0].user_id is not none
+        - iam_user_info.iam_users[0].user_name == '{{ test_user }}'
+
+  - name: ensure group exists
+    iam_group:
+      name: '{{ test_group }}'
+      users:
+        - '{{ test_user }}'
+      state: present
+    register: iam_group
+
+  - name: get info on IAM user(s) in group
```

Comment:
Might be good to add some additional tests adding a second user and making sure that

1. it doesn't show up when not added to the group
2. it does show up when added to the group
3. iam_user_info behaves properly when returning multiple values

(you'd be surprised at some of the dumb bugs I've seen over the years)

Reply:
https://github.com/ansible/ansible/pull/23382/commits/9e3eda56d319ccd6dcbbc1d7b513405cd1301e10 adds some test cases.